### PR TITLE
UINV-332 Include vendor primary address info

### PR DESCRIPTION
**Pull request by AndreiShumski2**
## Purpose
https://issues.folio.org/browse/UINV-332

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.
- [x] I've added appropriate record to the CHANGELOG.md
- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [x] If any API-related changes - okapi interfaces and permissions are reviewed/changed correspondingly
  - [x] There are no breaking changes in this PR.

If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/common/components/VendorPrimaryAddress/VendorPrimaryAddress.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      />
+    </>
+  );
+};
+
+VendorPrimaryAddress.propTypes = {
+  vendor: PropTypes.object,
+};
+
+export default VendorPrimaryAddress;
```
Comment (NikitaSedyx) [inline src/common/components/VendorPrimaryAddress/VendorPrimaryAddress.js:74]: we don't have a rule to prefer default export, please replace by named

File: src/common/hooks/useAddressCategories/useAddressCategories.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    () => {
+      return batchRequest(
+        ({ params: searchParams }) => ky.get(CATEGORIES_API, { searchParams }).json(),
+        address.categories,
+      );
+    },
+    { enabled: Boolean(address.categories?.length) },
+  );
+
+  const categoriesMap = data[0]?.categories?.reduce((acc, cat) => ({ ...acc, [cat.id]: cat.value }), {});
```
Comment (NikitaSedyx) [inline src/common/hooks/useAddressCategories/useAddressCategories.js:23]: why data[0] when you do batchRequest?

Comment (AndreiShumski2) [reply src/common/hooks/useAddressCategories/useAddressCategories.js:23]: it returns:
`[{ categories: [], totalRecords: 0 }]`

Comment (NikitaSedyx) [reply src/common/hooks/useAddressCategories/useAddressCategories.js:23]: `batchRequest` returns an array of responses, why do you limit to the first response only?

Comment (AndreiShumski2) [reply src/common/hooks/useAddressCategories/useAddressCategories.js:23]: got it :)

**Review by usavkov-epam — APPROVED** [review]

**Review by NikitaSedyx — APPROVED** [review]

**Pull request merged**